modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -288,27 +292,28 @@
           :week-of-year-us  (week-of-year column :us)
           :week-of-year-instance  (week-of-year column :instance)
           :month            (truncate :month)
-          :month-of-year    {$month column}
+          :month-of-year    (extract $month column)
           ;; For quarter we'll just subtract enough days from the current date to put it in the correct month and
           ;; stringify it as yyyy-MM Subtracting (($dayOfYear(column) % 91) - 3) days will put you in correct month.
           ;; Trust me.
           :quarter
           (if supports-dateTrunc?
             (truncate :quarter)
-            (mongo-let [#_{:clj-kondo/ignore [:unused-binding]} parts {:$dateToParts {:date column}}]
-                       {:$dateFromParts {:year  :$$parts.year
-                                         :month {$subtract [:$$parts.month
-                                                            {$mod [{$add [:$$parts.month 2]}
-                                                                   3]}]}}}))
+            (mongo-let [#_{:clj-kondo/ignore [:unused-binding]} parts {:$dateToParts {:date column :timezone (qp.timezone/results-timezone-id)}}]
+              {:$dateFromParts {:year  :$$parts.year
+                                :month {$subtract [:$$parts.month
+                                                   {$mod [{$add [:$$parts.month 2]}
+                                                          3]}]}
+                                :timezone (qp.timezone/results-timezone-id)}}))
 
           :quarter-of-year
-          {:$ceil {$divide [{$month column} 3.0]}}
+          {:$toInt {:$ceil {$divide [{$month {:date column :timezone (qp.timezone/results-timezone-id)}} 3.0]}}}
```

Comment:
```suggestion
          {:$toInt {:$ceil {$divide [(extract $month column) 3.0]}}}
```

Reply:
There is a similar calculation at line 289 which is using neither `extract` nor `:$toInt`.